pkg/api/store/cluster/cluster_store.go

```diff
@@ -633,6 +640,52 @@ func setBackupConfigSecretKeyIfNotExists(oldData, newData map[string]interface{}
 	}
 }
 
+func validateUpdatedS3Credentials(old, new map[string]interface{}) error {
+	newConfig := convert.ToMapInterface(values.GetValueN(new, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig", "s3BackupConfig"))
+	if newConfig == nil {
+		return nil
+	}
+	oldConfig := convert.ToMapInterface(values.GetValueN(old, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig", "s3BackupConfig"))
+	// remove "type" since it's added to the object by API, and it's not present in newConfig yet.
```

Comment:
Is it possible no s3 backup was enabled initially and oldConfig is nil, so you could directly call `validateS3Credentials` avoiding calling DeepEqual?